bindings/rascal/neighbourlist/structure_manager.py

```diff
@@ -228,8 +228,23 @@ def get_neighbourlist(structure, options):
 
 
 def convert_to_structure_list(frames):
-    if not isinstance(frames, Iterable):
+    """Convert an atomic structure to format used internaly in rascal.
+    Parameters
+    ----------
+    frames : ase.Atoms or list(ase.Atoms) or list(dict)
+        atomic structure(s) in various formats.
+        Note that the dictionary must be valid structure as per is_valid_structure
+    Returns
+    -------
+    structure_list : AtomicStructureList
+
+    """
+    if isinstance(frames, Atoms):
         frames = [frames]
+
+    assert isinstance(frames, list)
+    assert isinstance(frames[0], Atoms) or isinstance(frames[0], dict)
```

Comment:
```suggestion
    elif not isinstance(frames, Iterable):
        raise ValueError("Must pass either an ase.Atoms object or an iterable")
```
The check of whether the objects being iterated over are Atoms (or dict, or similar) is handled later on (see line 257 in this version).